File: ArkhamAsylum.Lib/Models/Diagnosis.cs
using Newtonsoft.Json;
using System;

namespace ArkhamAsylum.Lib.Models
{
    public class Diagnosis : Entity
    {
        public string Code { get; set; }
        public Guid DoctorId { get; set; }
        public Guid RecordId { get; set; }
        [JsonIgnore]
        public Doctor Doctor { get; set; }
        [JsonIgnore]
        public Record Record { get; set; }
    }
}

File: ArkhamAsylum.Web/wwwroot/Scripts/Models/Diagnosis.js
class Diagnosis extends Entity
{
    constructor(json)
    {
        super(json)
        if (json)
        {
            this.Code = json.code;
            this.DoctorId = json.doctorId;
            this.RecordId = json.recordId;
        }
        else
        {
            this.Code = "";
            this.DoctorId = "00000000-0000-0000-0000-000000000000";
            this.RecordId = "00000000-0000-0000-0000-000000000000";
        }
    }
}